File: algorithms-learning/leetcode-study/python/1-two-sum.py
from typing import Dict, List


class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(i + 1, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]
        return []


class Solution1:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashtable = dict()
        for i in range(len(nums)):
            if target - nums[i] in hashtable.keys():
                return [hashtable.get(target - nums[i]), i]
            hashtable[nums[i]] = i
        return []


# test case
nums = [2, 2, 11, 15]
target = 4
s = Solution1()
print(s.twoSum(nums, target))

File: algorithms-learning/leetcode-study/cpp/1-two-sum.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution
{
public:
    vector<int> twoSum(vector<int> &nums, int target)
    {`
        int n = nums.size();
        for (int i = 0; i < n; i++)
        {
            for (int j = i + 1; j < n; j++)
            {
                if (nums[i] + nums[j] == target)
                {
                    return {i, j};
                }
            }
        }
        return {};
    }
};

int main()
{
    Solution solution;
    vector<int> nums{2, 3, 5, 7};
    vector<int> result = solution.twoSum(nums, 5);
    for (int i: result){
        cout << i << endl;
    }
    return 0;
}